components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/validators/JDBCPermissionBasedInternalScopeValidator.java

```diff
@@ -129,18 +128,22 @@
         return scopesToRespond.toArray(new String[0]);
     }
 
-    private String[] getRequestedScopes(String[] scopes) {
+    public String[] validateScope(String[] requestedScopes, AuthenticatedUser authenticatedUser, String clientId) {
```

Comment:
let's add method comments for public methods

Reply:
also, the method name is a bit confusing because what we are doing here AFAIU is returning the set of scopes that are requested? please correct me if I'm wrong.

Reply:
`let's add method comments for public methods`
Hi @chamathns, This is another overloaded method to validate the scope. Since there weren't any method comments in the previous two methods, I didn't add a method comment for this new method as well
https://github.com/wso2-extensions/identity-inbound-auth-oauth/blob/8b9e7fd118c334089b498050764f2053e7af9807/components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/validators/JDBCPermissionBasedInternalScopeValidator.java#L81
https://github.com/wso2-extensions/identity-inbound-auth-oauth/blob/8b9e7fd118c334089b498050764f2053e7af9807/components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/validators/JDBCPermissionBasedInternalScopeValidator.java#L105

Reply:
ideally, all public methods should have their own method descriptions or else, an `inherit doc` reference if they are overridden methods

Reply:
As you said, `getRequestedScopes` (This method moved to a different class) will be used to get a set of scopes that are requested. But `validateRequest` method is used to validate scopes that are relevant to the authenticated user. These are two different methods. But GitHub shows that I changed the `getRequestedScopes` into `validateRequest` method.